DotNetBuildFromSource fail to compile src/libraries/shims/System.Data/ref/System.Data.csproj

Our internal CI builds have been failing since mid May (d0740fc967d8b3a9b215a24ca0a007a9602a9fa9 working, d7a30ba1f1c9ccce26a6a223898b7b9aba186a3a failing).

The build error is:
```
/Microsoft.Common.CurrentVersion.targets(2353,5): error MSB3106: Assembly strong name "/home/tester/.nuget/packages/system.data.sqlclient/4.8.5/ref/net6.0/System.Data.SqlClient.dll" is either a path which could not be found or it is a full assembly name which is badly formed. If it is a full assembly name it may contain characters that need to be escaped with backslash(\). Those characters are Equals(=), Comma(,), Quote("), Apostrophe('), Backslash(\). [/home/tester/runtime/src/libraries/shims/System.Data/ref/System.Data.csproj]
```
I've extracted the `System.Data.SqlClient 4.8.5` package from nuget.org. There is no `ref/net6.0` present.

I can reproduce the failure locally when running the build with (`DotNetBuildFromSource`).

@ViktorHofer this probably regressed by https://github.com/dotnet/runtime/pull/84782.

cc @omajid

--- a/src/libraries/shims/System.Data/Directory.Build.props
+++ b/src/libraries/shims/System.Data/Directory.Build.props
@@ -7,7 +7,7 @@
     <StrongNameKeyId>ECMA</StrongNameKeyId>
     <SystemDataSqlClientTFM>netcoreapp2.1</SystemDataSqlClientTFM>
     <!-- Keep in sync with TFM exposed in source-build-reference-packages/src/referencePackages/src/system.data.sqlclient/4.8.5 -->
-    <SystemDataSqlClientTFM Condition="'$(DotNetBuildFromSource)' == 'true'">net6.0</SystemDataSqlClientTFM>
+    <SystemDataSqlClientTFM Condition="'$(ArcadeBuildFromSource)' == 'true'">net6.0</SystemDataSqlClientTFM>
   </PropertyGroup>
 
   <ItemGroup>

